omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/PatientIdentifierController1_9Test.java

```diff
@@ -125,6 +126,23 @@ public void shouldVoidAPatientIdentifier() throws Exception {
 	}
 	
 	@Test
+	public void shouldListAllPatientIdentifiersWithVoidedIdentifiersForAPatient() throws Exception {
+		assertEquals(false, service.getPatientIdentifierByUuid(getUuid()).isVoided());
+		MockHttpServletRequest req = request(RequestMethod.DELETE, getURI() + "/" + getUuid());
+		final String reason = "none";
+		req.addParameter("reason", reason);
+		handle(req);
+		assertEquals(true, service.getPatientIdentifierByUuid(getUuid()).isVoided());
+		SimpleObject result = deserialize(handle(newGetRequest(getURI(), new Parameter(
+		        RestConstants.REQUEST_PROPERTY_FOR_INCLUDE_ALL, "true"))));
+		assertNotEquals(getAllCount(), Util.getResultsSize(result));
+		
+		SimpleObject nextResult = deserialize(handle(newGetRequest(getURI(), new Parameter(
+		        RestConstants.REQUEST_PROPERTY_FOR_INCLUDE_ALL, "false"))));
```

Comment:
Isn't getAllCount() the one that has all, including the voided ones?

Reply:
No. Get all returns only the active Identifiers 

>return service.getPatientByUuid(RestTestConstants1_8.PATIENT_UUID).getActiveIdentifiers().size();

That is why getAllCount() returns 1 after we void an identifier instead of two but REQUEST_PROPERTY_FOR_INCLUDE_ALL returns 2 including the voided.

Reply:
Oh i see! 😊

Reply:
Yes 😄  Is there any other thing I should do?